gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+
+## Rationale and Alternatives
+
+Alternative (possibly simpler) approaches have been suggested, including:
+
+- Could the Gateway just pre-pay a fixed amount per subgraph?
+- Could Indexers just charge a specific (Indexer-determined) amount for each subgraph in arrears?
+
+However, the design of The Graph protocol must be done with an adversarial mindset. We might assume that Indexers will try to charge honestly to maximize the indexing agreements they receive in the future. But the protocol must be safe against malicious participants that may not care about this. For example, a malicious set of Indexers might be shorting GRT while griefing several users, trying to profit from discrediting the protocol. Or they may have a large time preference for money, so they extract the maximum payment and don't care about future agreements. The typical cryptoeconomic approach to address these risks is to establish a cost-of-corruption (through slashing), so that malicious actors are punished and participants can at least quantify the risk in each interaction.
+
+In the suggested solutions described above (a fixed amount per subgraph, or an amount determined by the Indexer in arrears), there are a few potential attack vectors to consider: the Gateway could refuse payment after the fact, or the Indexer could take the money and run, or the Indexer could try to charge a ridiculous amount after the Gateway has promised payment.
+
+Considering this, the protocol is designed to minimize the need for trust, in two directions:
+
+- Gateways should not need to trust that Indexers will do their job or provide correct data. Otherwise, Gateways would only query trusted Indexers (e.g. Indexers with a reputation or direct relationship with the Gateway, which could lead to centralization or a winner-take-all Indexer).
+- Indexers should not need to trust that Gateways will provide payments after completing their job. Otherwise, Indexers would only serve trusted Gateways to ensure payment. For example, Indexer would only serve reputable Gateways, which could lead to Gateway centralization and force consumers to access the network through the set of trusted Gateways.
+
+The proposed payment system minimizes trust in both directions.
+
+When combined with the cryptoeconomic security from stake and appropriate fault-detection, the protocol can establish a cost-of-corruption, or a cost to faults in general. With this in place, a new participant entering the Network, for example as a Gateway or even as a Consumer running their own Gateway, can use the Network to its full potential. Any trusted relationship enshrined at the protocol level would directly jeopardize the Network's censorship-resistance and resilience, as it could require participants to interact with only a handful of trusted participants.
```

Comment:
I think either 'the network' or 'The Network', because 'the' is part of the name as I see it, if used as a noun.
```suggestion
When combined with the cryptoeconomic security from stake and appropriate fault-detection, the protocol can establish a cost-of-corruption, or a cost to faults in general. With this in place, a new participant entering the network, for example as a Gateway or even as a Consumer running their own Gateway, can use the network to its full potential. Any trusted relationship enshrined at the protocol level would directly jeopardize the network's censorship-resistance and resilience, as it could require participants to interact with only a handful of trusted participants.
```